### PR TITLE
User services

### PR DESCRIPTION
**Pull request by JosephOri**
added user services to the frontend and implemented the functionallity of `LoginPage.jsx` and `SignupPage.jsx`

### CODE REVIEW

File: src/components/Graph/Graph.jsx [hunk 1]
```diff
@@ -1,7 +1,6 @@
-import React from 'react';
 import Node from './Node';
 import Edge from './Edge';
-import List from '../common/List';
+import List from '../Common/List';
```
Comment (Tamir198) [inline src/components/Graph/Graph.jsx:4]: Please discuss the other team members before changing folder name (this should be done between sprints to avoid conflicts)

File: src/components/Graph/GraphPreview.jsx [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
-        height:200,  backgroundColor:"lightpink"}}> graph need to be dispalyed here</Card>
-        
-        </div>
-    
-    );
+const GraphPreview = ({ key }) => {
+  return (
+    <div className="graph-preview">
+      <Card
+        title={texts.GRAPH_PREVIEW}
```
Comment (Tamir198) [inline src/components/Graph/GraphPreview.jsx:9]: Same thing in here - this is not related to this pr

File: src/components/Login.jsx [hunk 3]
```diff
@@ -1,39 +1,53 @@
...
+      e.preventDefault();
+      const response = await axios.post(BACKEND_LOGIN_URL, { name, password });
+      const token = response.data;
+      localStorage.setItem('accessToken', token.accessToken);
+      localStorage.setItem('refreshToken', token.refreshToken);
+      alert('Login successful');
+      setName('');
+      setPassword('');
+      window.location.href = '/StartPage';
+    } catch (error) {
```
Comment (Tamir198) [inline src/components/Login.jsx:23]: If this is for page navigation you should use `useNavigate` coming from react router

File: src/components/Signup.jsx [hunk 4]
```diff
@@ -77,52 +34,86 @@ const tailFormItemLayout = {
...
+    setSignupFormData({ ...signupFormData, email: e.target.value });
+  };
+
+  const handlePasswordChange = (e) => {
+    setSignupFormData({ ...signupFormData, password: e.target.value });
+  };
+
+  const handleConfirmPasswordChange = (e) => {
+    setSignupFormData({ ...signupFormData, confirmPassword: e.target.value });
+  };
```
Comment (Tamir198) [inline src/components/Signup.jsx:49]: You could reduce code duplication with a shared function, for example in this pseudo code : 

```js 
const handleInputChange = (fieldName, value) => {
  setSignupFormData({ ...signupFormData, [fieldName]: value });
};
```

File: src/components/Signup.jsx [hunk 5]
```diff
@@ -77,52 +34,86 @@ const tailFormItemLayout = {
...
+
+  const isFormDataValidCheck = () => {
+    const { name, email, password, confirmPassword } = signupFormData;
+    const isAllFieldsFilled = name && email && password && confirmPassword;
+    const isEmailValid = email.includes('@') && email.includes('.');
+    if (!isAllFieldsFilled) {
```
Comment (Tamir198) [inline src/components/Signup.jsx:55]: I would replace this with a regex because checking for `@` and `.` is not enough for email validation (for example `@.` would be a valid mail

File: src/utils/allowCorsForAxios.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+const allowCorsForAxios = (axios) => {
+  axios.defaults.headers.common['Access-Control-Allow-Origin'] = '*';
+};
+
+export default allowCorsForAxios;
```
Comment (Tamir198) [inline src/utils/allowCorsForAxios.js:5]: Are you using this? If not remove this

**Review by Tamir198 — COMMENTED** [review]
Why did you added the `createGraph` file? this is not related to this pr.
Also please remove your prints.

File: src/utils/allowCorsForAxios.js (hunk 6 above)
Comment (JosephOri) [reply src/utils/allowCorsForAxios.js:5]: I am using this in both login and signup pages, I have been taught to write functions with a clear name insted of comments
